src/Controller.ts

```diff
@@ -20,47 +18,36 @@ export class Controller {
-        const mainClassList = await vscode.commands.executeCommand('java.execute.workspaceCommand', 'vscode.java.resolveMainClass', app.path);
-        if (mainClassList && mainClassList instanceof Array && mainClassList.length > 0) {
-            const mainClassData: MainClassData = mainClassList.length === 1 ? mainClassList[0] :
-                await vscode.window.showQuickPick(mainClassList.map(x => Object.assign({ title: x.mainClass }, x)));
+        if (mainClasData) {
+            let targetConfig = this._getLaunchConfig(mainClasData);
+            if (!targetConfig) {
+                targetConfig = await this._createNewLaunchConfig(mainClasData);
+            }
+            // TO REMOVE:
+            // This is a workaround to make "start" work. For Java Debugger, "Start without debugging" feature 
+            // has not been released, it fallbacks to "Start debugging" mode.
+            // See: https://github.com/Microsoft/vscode-java-debug/issues/351
+            debug ? await this._enableAllBPs() : await this._disableAllBPs();
 
-            outputChannel.clear();
-            outputChannel.show();
-            let stderr: string = '';
-            const javaProcess: ChildProcess = spawn("java", ["-classpath", classpathString, mainClassData.mainClass]);
-            javaProcess.stdout.on('data', (data: string | Buffer): void => {
-                outputChannel.append(data.toString());
-            });
-            javaProcess.stderr.on('data', (data: string | Buffer) => {
-                stderr = stderr.concat(data.toString());
-                outputChannel.append(data.toString());
-            });
-            javaProcess.on('error', (err: Error) => {
-                console.error("on error: ", err.toString());
-            });
-            javaProcess.on('exit', (code: number, signal: string) => {
-                this._setState(app, STATE_INACTIVE);
-            });
-            app.process = javaProcess;
+            const ok: boolean = await vscode.debug.startDebugging(
+                vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(app.path)),
+                Object.assign({}, targetConfig, { noDebug: !debug })
+            );
+            if (ok) {
+                app.activeSession = vscode.debug.activeDebugSession;
+                this.setState(app, STATE_RUNNING);
+            }
         } else {
             vscode.window.showWarningMessage("Found no Main Class.");
         }
     }
 
-    public async stopBootApp(app: BootApp): Promise<void> {
-        // TODO: How to send a shutdown signal to the app instead of killing the process directly?
-        if (app.process) {
-            app.process.kill("SIGTERM");
-            if (app.process.killed) {
-                app.process = undefined;
-            }
+    public async stopBootApp(app: BootApp, restart?: boolean): Promise<void> {
+        if (app.activeSession) {
```

Comment:
I suggest keeping this TODO. It is still valid/desirable.

Reply:
Sure, it was mistakenly deleted.